src/transform/class/inheritance/index.js

```diff
@@ -0,0 +1,34 @@
+import UtilInherits from './UtilInherits';
+import Prototypal from './Prototypal';
+
+/**
+ * Processes nodes to detect super classes and return information for later
+ * transformation.
+ */
+export default class Inheritance {
+  /**
+   * @param {Object} cfg
+   *   @param {PotentialClass[]} cfg.potentialClasses Class name
+   */
+  constructor({potentialClasses}) {
+    this.potentialClasses = potentialClasses;
+    this.utilInherits = new UtilInherits(this);
+    this.prototypal = new Prototypal(this);
```

Comment:
It looks like `UtilInherits` and `Prototypal` make some use of the `Inheritance` class instance, but turns out they only extract the `potentialClasses` field out of it. Better to pass `potentialClasses` to them directly.